tests/ui-fulldeps/hash-stable-is-unstable.stderr

```diff
@@ -6,7 +6,6 @@ LL | extern crate rustc_data_structures;
    |
    = note: see issue #27812 <https://github.com/rust-lang/rust/issues/27812> for more information
    = help: add `#![feature(rustc_private)]` to the crate attributes to enable
-   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date
```

Comment:
I looked through the call stack for some time, and I'm not sure why this changed... It does seem like there's places in the compiler where these messages are added directly without using the relevant functions (should I worry about that case?) but I don't think this is one of them

Reply:
it's okay if this error doesn't get the note, it will never be stable

Reply:
Here's an example of a place where the message is being directly, maybe this is another case of something that will never be stable?

https://github.com/rust-lang/rust/blob/master/compiler/rustc_const_eval/src/transform/check_consts/ops.rs#L316-L334

Reply:
I think it's important to understand the shortcomings of the solution, but to be honest this probably isn't the right time to refactor these diagnostics

Reply:
This is definitely a call site where the diagnostic would be very relevant, as library functions get made const stable all the time. But yes, not in this PR,happy to take a follow-up (by leaving the issue open with a note about this).